core/chains/evm/logpoller/parser.go

```diff
@@ -0,0 +1,493 @@
+package logpoller
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+)
+
+const (
+	sequenceFieldAlias = "sequence"
+	sequenceField      = "CONCAT(blockHash, '-', txHash, '-', logIndex)"
+	blockFieldName     = "block_number"
+	timestampFieldName = "timestamp"
+	txHashFieldName    = "tx_hash"
+	eventSigFieldName  = "event_sig"
+	chainIDFieldName   = "evm_chain_id"
+	evmLogsTableName   = "evm.logs"
+)
+
+type dslQueryProps struct {
+	ChainID *big.Int
+	Filter  query.KeyFilter
+	Limiter query.LimitAndSort
+}
+
+func (p *dslQueryProps) hasCursorLimit() bool {
+	return p.Limiter.Limit.Cursor != nil && p.Limiter.Limit.CursorDirection != nil
+}
+
+func (p *dslQueryProps) hasSequenceSort() bool {
+	for _, order := range p.Limiter.SortBy {
+		switch order.(type) {
+		case query.SortBySequence:
+			return true
+		default:
+			continue
+		}
+	}
+
+	return false
+}
+
+// pgDSLParser is a visitor that builds a postgres query and arguments from a commontypes.QueryFilter
+type pgDSLParser struct {
+	chainID    *big.Int
+	args       *namedArgs
+	expression string
+}
+
+var _ query.Visitor = (*pgDSLParser)(nil)
+
+func (v *pgDSLParser) ComparerPrimitive(p query.ComparerPrimitive) {}
+
+func (v *pgDSLParser) BlockPrimitive(p query.BlockPrimitive) {
+	v.expression = fmt.Sprintf(
+		"%s %s :%s",
+		blockFieldName,
+		cmpOpToString(p.Operator),
+		v.args.Add(blockFieldName, p.Block),
+	)
+}
+
+func (v *pgDSLParser) ConfirmationPrimitive(p query.ConfirmationsPrimitive) {
+	switch p.ConfirmationLevel {
+	case query.Finalized:
+		v.expression = v.nestedConfQuery(true, 0)
+	case query.Unconfirmed:
+		fallthrough
+	default:
+		v.expression = v.nestedConfQuery(false, uint64(types.Unconfirmed))
+	}
+}
+
+func (v *pgDSLParser) TimestampPrimitive(p query.TimestampPrimitive) {
+	v.expression = fmt.Sprintf(
+		"%s %s :%s",
+		timestampFieldName,
+		cmpOpToString(p.Operator),
+		v.args.Add(timestampFieldName, p.Timestamp),
+	)
+}
+
+func (v *pgDSLParser) TxHashPrimitives(p query.TxHashPrimitive) {
+	v.expression = fmt.Sprintf(
```

Comment:
Not sure where is the best place to validate thath txHash is common.Hash. Maybe we should have evm specific filter that this gets remapped to? Same as ValuComparer?